demo/src/pages/ui.rs

```diff
@@ -0,0 +1,293 @@
+
+        div { class: "space-y-6",
+          // default Textarea
+          Label {
+            label_text: Some("Default Textarea".into()),
+            Textarea {
+              value: text_area_value.read().to_string(),
+              on_change: move |value| text_area_value.set(value),
+              placeholder: Some("Enter text here...".into()),
+              class: "w-full p-4 border rounded-lg border-gray-300 focus:outline-none focus:ring-2 focus:ring-primary"
+            }
+          }
+
+          // disabled Textarea
+          Label {
+            label_text: Some("Disabled Textarea".into()),
+            Textarea {
+              value: "Disabled content".to_string(),
+              disabled: true,
+              placeholder: Some("Cannot edit this text...".into()),
+              class: "w-full p-4 border rounded-lg bg-gray-200 text-gray-500 cursor-not-allowed"
+            }
+          }
+
+          // textarea with on_enter functionality
+          Label {
+            label_text: Some("Textarea with Enter Handler".into()),
+            Textarea {
+              value: text_area_value.read().to_string(),
+              on_change: move |value| text_area_value.set(value),
+              on_enter: move |value| entered_text.set(value),
+              placeholder: Some("Type and press Shift+Enter...".into()),
+              class: "w-full p-4 border rounded-lg border-gray-300 focus:outline-none focus:ring-2 focus:ring-primary"
+            }
+          }
+          p { class: "text-sm text-gray-600 italic",
+            "Last entered text: {entered_text.read()}"
+          }
+
+          // textarea with custom styles
+          Label {
+            label_text: Some("Custom Styled Textarea".into()),
+            Textarea {
+              value: text_area_value.read().to_string(),
+              on_change: move |value| text_area_value.set(value),
+              placeholder: Some("Styled input...".into()),
+              class: "w-full p-4 border-2 border-dashed border-primary text-primary focus:outline-none focus:ring-2 focus:ring-primary",
+              style: "padding: 1rem; font-style: italic;"
+            }
+          }
+
+          // loading Spinner
+          div { class: "flex items-center gap-4",
+            FreeIconSpinner { size: 24 }
+            span { class: "text-sm text-gray-600", "Loading..." }
+          }
+        }
+      }
+    }
+  }
+}
+
+#[derive(Props, Clone, PartialEq)]
+pub struct ComponentSectionUIProps {
+  title: &'static str,
+  description: &'static str,
+  children: Element,
+}
+
+#[component]
+fn ComponentSectionUI(props: ComponentSectionUIProps) -> Element {
+  rsx! {
+    section { class: "mb-12",
+      h2 { class: "text-2xl font-semibold mb-2", {props.title} }
+      p { class: "text-gray-600 mb-6", {props.description} }
+      div { class: "bg-white p-6 rounded-lg shadow-sm border", {props.children} }
+    }
+  }
+}
```

Comment:
i've noticed, you use such components on different pages - the structure is the same in general, and you redeclare them each time. Maybe, make one such component and make it reusable for all pages?

Reply:
Move it in the components folder please, let's there will be all reusable parts